opentracing-spring-web-starter/src/main/java/io/opentracing/contrib/spring/web/starter/WebClientTracingAutoConfiguration.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2016-2019 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.web.starter;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.tracer.configuration.TracerAutoConfiguration;
+import io.opentracing.contrib.spring.web.client.TracingWebClientBeanPostProcessor;
+import io.opentracing.contrib.spring.web.client.WebClientSpanDecorator;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+
+/**
+ * Instrumentation of {@link WebClient}. It also instruments {@link WebClient.Builder} and all instances created with it.
+ *
+ * @author Csaba Kos
+ */
+@Configuration
+@ConditionalOnBean(Tracer.class)
+@ConditionalOnClass(WebClient.class)
+@ConditionalOnProperty(prefix = WebClientTracingProperties.CONFIGURATION_PREFIX, name = "enabled", matchIfMissing = true)
+@AutoConfigureAfter(TracerAutoConfiguration.class)
+@EnableConfigurationProperties(WebClientTracingProperties.class)
```

Comment:
Should `WebClientTracingProperties` be renamed? I mean the name is a little misleading since it's configuration for `RestTemplate`, `AsyncRestTemplate` and `WebClient`.
Also, the comments in `WebClientTracingProperties` should be updated to reflect this fact

Reply:
I don't have an opinion about renaming - I could see arguments both pro and con. I'll leave the decision up to you and @pavolloffay. (Because the class is `public`, it is in theory part of the public API, so that'd be one reason for not renaming it.)

I will update the comments though.

Reply:
It's true that it's public, public configuration classes would rarely be used directly in application code. I would assume that at least a minor version bump would be warranted even without the rename, so some minor breaking could probably be expected